x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/types.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export enum CrawlerPolicies {
+  Allow = 'allow',
+  Deny = 'deny',
+}
+
+export enum CrawlerRules {
+  beginsWith = 'begins',
+  endsWith = 'ends',
+  contains = 'contains',
+  regex = 'regex',
+}
+
+export interface CrawlRule {
+  id: string;
+  policy: CrawlerPolicies;
+  rule: CrawlerRules;
+  pattern: string;
+}
+
+export interface EntryPoint {
+  id: string;
+  value: string;
+}
+
+export interface Sitemap {
+  id: string;
+  url: string;
+}
+
+export interface CrawlerDomain {
+  createdOn: string;
+  documentCount: number;
+  id: string;
+  lastCrawl?: string;
+  url: string;
+  crawlRules: CrawlRule[];
+  defaultCrawlRule?: CrawlRule;
+  entryPoints: EntryPoint[];
+  sitemaps: Sitemap[];
+}
+
+export interface CrawlerDomainFromServer {
+  id: string;
+  name: string;
+  created_on: string;
+  last_visited_at?: string;
```

Comment:
Why haven't I had to do deal with snake casing vs camel casing on any other API endpoints? 🤔

Reply:
I think because this is a customer/user facing endpoint which uses snake_case, while most of our internal endpoints use camelCase

Reply:
Ah, got it.

You don't have to solve this here, but if we'll need to do a bunch of these it'd be cool to have a utility (on the client or logic) that converts all of the keys in an object to cameCase automatically. We can cross that bridge when we get there though.